src/components/appList.js

```diff
@@ -69,25 +67,69 @@ const Description = styled.p`
     color: ${selectedTheme.accentColor};
 `;
 
-const appList = () => (
-    <AppListContainer>
-        <ApplicationsText>Applications</ApplicationsText>
-        <AppsContainer>
-        {
-            appData.apps.map((app) => (
-                <AppContainer>
-                    <IconContainer>
-                        <MaterialIcon icon={app.icon} color={selectedTheme.mainColor}/>
-                    </IconContainer>
-                    <AppDetails>
-                        <Link href={app.URL}>{app.name}</Link>
-                        <Description>{app.displayURL}</Description>
-                    </AppDetails>
-                </AppContainer>
-            ))
-        }
-        </AppsContainer>
-    </AppListContainer>
-);
-
-export default appList;
\ No newline at end of file
+const ErrorMessage = styled.p`
+    color: red;
+`;
+
+function handleResponse(response) {
+    if (response.ok) {
+        return response.json();
+    }
+    throw new Error('Failed to load app data.');
+}
+
+function useAppData() {
+    const [appData, setAppData] = useState({ apps: [], error: false });
+    const fetchAppData = useCallback(() => {
+        (process.env.NODE_ENV === 'production'
+            ? fetch('/apps.json').then(handleResponse)
+            : import('./data/apps.json')
```

Comment:
The idea here is that if a developer is just testing, they can still use the original json file which still supports HMR. In production, switch over to an HTTP request to load in JSON so it can be refreshed as-needed without rebuilding your entire application.